kernel/Kconfig.vm

```diff
@@ -100,7 +100,7 @@ endif # KERNEL_VM_SUPPORT
 
 menuconfig MMU
 	bool "MMU features"
-	depends on CPU_HAS_MMU
+	depends on CPU_HAS_MMU && (!X86 || (X86 && X86_MMU))
```

Comment:
There is nothing wrong with X86 here, the same issue applies to Arm and others, MMU should only be enabled by the mmu arch specific implementation and not directly.

Reply:
It it's only meant to be enabled by the architecture, then why is this a user-configurable option?

Similarly, why do we have e.g. CONFIG_I2C or any hardware or architecture level Kconfig options that users can choose to enable or disable?

Why is it that an application can choose to enable MMU in one case but cannot choose to enable MMU in another case?

If an application chooses to enable the standard POSIX system interfaces, then they typically want the features that would provide them.